heat/core/vmap.py

```diff
@@ -0,0 +1,88 @@
+"""
+This implements a functionality similar to PyTorchs vmap function.
+"""
+
+import torch
+from .dndarray import DNDarray
+from .factories import array
+from .communication import MPI_WORLD
+
+__all__ = ["vmap"]
+
+
+def vmap(func, out_dims=0, randomness="error", *, chunk_size=None):
+    """
+    This function is used to apply a function to a DNDarray in a vectorized way.
+    `heat.vmap` return a callable that can be applied to DNDarrays.
+    Vectorization will automatically take place along the split axis/axes; therefore, unlike in PyTorch, there is no argument `in_dims`.
```

Comment:
Connection between "batch axis" and "split axis". Do all parameters need to have the same split axis?

Reply:
added corresponding comment to docstring